### PR TITLE
Feat/policies

### PR DESCRIPTION
**Pull request by pheel**
## Feat/policies

- Updated ProjectChat.jsx's instance check method
- Updated NLU training method

### CODE REVIEW

File: botfront/imports/api/instances/instances.methods.js [hunk 1]
```diff
@@ -26,10 +31,84 @@ export const createInstance = async (project) => {
...
+    const mappingTriggers = policies
+        .filter(policy => policy.name.includes('BotfrontMappingPolicy'))
+        .map(policy => policy.triggers.map(trigger => trigger.action))
+        .reduce((coll, curr) => coll.concat(curr), [])
+        .join('\n  - ');
+    const mappingStory = `## mapping story\n* mapping_intent\n  - action_botfront_mapping_follow_up\n  - ${mappingTriggers}`;
+    const storyGroups = StoryGroups.find(
+        { projectId },
+        { stories: 1 },
+    ).fetch().map(group => group.stories.join('\n'));
```
Comment (MatthieuJnon) [inline botfront/imports/api/instances/instances.methods.js]: This wont work as group.stories is an array of objects

File: botfront/imports/api/instances/instances.methods.js [hunk 2]
```diff
@@ -182,23 +265,24 @@ if (Meteor.isServer) {
                     error = new Meteor.Error('500', e);
                 }
 
-                console.log(error);
                 throw error;
             }
         },
+
+        'extractDomainFromStories'(storyGroup) {
```
Comment (MatthieuJnon) [inline botfront/imports/api/instances/instances.methods.js]: What's the purpose of this method ? It's never called from anywhere except in tests

**Comment by MatthieuJnon** [comment]
Make sure to run all tests once you're done with request changes

File: botfront/imports/ui/components/project/ProjectChat.jsx [hunk 3]
```diff
@@ -145,15 +119,24 @@ class ProjectChat extends React.Component {
                         </Menu.Item>
                     </Menu.Menu>
                 </Menu>
-                {socketUrl && path && (
+                {selectedLanguage === '' && (
```
[the remark this answers is not in the document]
Comment (pheel) [reply botfront/imports/ui/components/project/ProjectChat.jsx]: Sorry-- didn't see there was work being done on that file.

File: botfront/imports/api/instances/instances.methods.js (hunk 1 above)
Comment (pheel) [reply botfront/imports/api/instances/instances.methods.js]: fixed

File: botfront/imports/api/instances/instances.methods.js (hunk 2 above)
Comment (pheel) [reply botfront/imports/api/instances/instances.methods.js]: There was no other way of testing at the time.

File: botfront/imports/ui/components/project/ProjectChat.jsx [hunk 4]
```diff
@@ -20,37 +20,9 @@ class ProjectChat extends React.Component {
...
-            'instance.findByType',
-            projectId,
-            'core',
-            wrapMeteorCallback((err, item) => {
-                if (!item) {
-                    this.setState({ noCore: true });
-                    return;
-                }
-                // eslint-disable-next-line no-useless-escape
-                const matches = item.host.match(/^https?:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
```
[the remark this answers is not in the document]
Comment (pheel) [reply botfront/imports/ui/components/project/ProjectChat.jsx]: My bad~!

**Pull request merged**